api/cloudspecificextension/deployments_api.go

```diff
@@ -0,0 +1,121 @@
+func (csets *CloudSpecificExtensionDeploymentService) ListDeployments() (deployments []*types.CloudSpecificExtensionDeployment, err error) {
+	log.Debug("ListDeployments")
+
+	data, status, err := csets.concertoService.Get("/cse/deployments")
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &deployments); err != nil {
+		return nil, err
+	}
+
+	return deployments, nil
+}
+
+// GetDeployment returns a cloud specific extension deployment by its ID
+func (csets *CloudSpecificExtensionDeploymentService) GetDeployment(deploymentID string) (deployment *types.CloudSpecificExtensionDeployment, err error) {
+	log.Debug("GetDeployment")
+
+	data, status, err := csets.concertoService.Get(fmt.Sprintf("/cse/deployments/%s", deploymentID))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &deployment); err != nil {
+		return nil, err
+	}
+
+	return deployment, nil
+}
+
+// CreateDeployment creates a cloud specific extension deployment
+func (csets *CloudSpecificExtensionDeploymentService) CreateDeployment(templateID string, deploymentVector *map[string]interface{}) (deployment *types.CloudSpecificExtensionDeployment, err error) {
+	log.Debug("CreateDeployment")
+
+	data, status, err := csets.concertoService.Post(fmt.Sprintf("/cse/templates/%s/deployments", templateID), deploymentVector)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &deployment); err != nil {
+		return nil, err
+	}
+
+	return deployment, nil
+}
+
+// UpdateDeployment updates a cloud specific extension deployment by its ID
+func (csets *CloudSpecificExtensionDeploymentService) UpdateDeployment(deploymentVector *map[string]interface{}, deploymentID string) (deployment *types.CloudSpecificExtensionDeployment, err error) {
+	log.Debug("UpdateDeployment")
+
+	data, status, err := csets.concertoService.Put(fmt.Sprintf("/cse/deployments/%s", deploymentID), deploymentVector)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &deployment); err != nil {
+		return nil, err
+	}
+
+	return deployment, nil
+}
+
+// DeleteDeployment deletes a cloud specific extension deployment by its ID
+func (csets *CloudSpecificExtensionDeploymentService) DeleteDeployment(deploymentID string) (err error) {
+	log.Debug("DeleteDeployment")
+
+	data, status, err := csets.concertoService.Delete(fmt.Sprintf("/cse/deployments/%s", deploymentID))
+	if err != nil {
+		return err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Deleting a CSE deployment just starts the undeploy process: you should return here a representation parsed from the response